tests/test_resource.py

```diff
@@ -162,15 +167,42 @@ class List(object):
         self.assertIn(mock.sentinel.acl, acls_args)
         self.assertIn(mock.sentinel.collection_acl, acls_args)
 
+    @mock.patch('cornice.resource.Service')
+    def test_factory_is_autowired(self, mocked_service):
+        @resource(collection_path='/list', path='/list/{id}', name='list')
+        class List(object):
+            pass
+        factory_args = [kw.get('factory') for _, kw in mocked_service.call_args_list]
+        self.assertEqual([List, List], factory_args)
+
+    @mock.patch('cornice.resource.Service')
+    def test_acl_is_discarded(self, mocked_service):
```

Comment:
As seen earlier, instead of silently discard it, maybe we should raise explicitly?